### PR TITLE
Custom environment tutorial

### PR DESCRIPTION
**Pull request by RedTachyon**
Tutorial about using a custom environment with `imitation`, see #718

### CODE REVIEW

File: docs/tutorials/8_train_custom_env.ipynb
Comment (AdamGleave) [inline docs/tutorials/8_train_custom_env.ipynb:10]: This might give the impression that `imitation` only supports training policies in custom environments, and not training rewards, when in fact it supports both (although behavioral cloning can only learn policies).

File: docs/tutorials/8_train_custom_env.ipynb
Comment (AdamGleave) [inline docs/tutorials/8_train_custom_env.ipynb:43]: This is quite a complex environment just for a tutorial, a user could get lost in some of these details. Consider something even more toy?

File: docs/tutorials/8_train_custom_env.ipynb
Comment (AdamGleave) [inline docs/tutorials/8_train_custom_env.ipynb:271]: This is a good illustration! I like the "There Is More Than One Way To Do It" style (while still suggesting the approach that we think is best for common use cases).

File: docs/tutorials/8_train_custom_env.ipynb
Comment (AdamGleave) [inline docs/tutorials/8_train_custom_env.ipynb:358]:
```suggestion
    "Keep in mind that while we're using BC in this tutorial, you can just as easily use any of the other algorithms with the environment prepared in this way."
```

**Review by AdamGleave — APPROVED** [review]
LGTM -- a few minor comments/suggested changes, but should be fine to merge once you've addressed them (and CI checks are passing).

**Comment by RedTachyon** [comment]
Addressed all the comments, changed the environment to a simpler observation matching one

**Comment by AdamGleave** [comment]
Closing in favor of #746 to make one tweak to the wording (don't have push access on this branch.)

**Pull request closed without merging**
